### DISCUSSION TITLE
File upload problem in dart. code: validation_unknown_filenames, message: The field contains unknown filenames.

### DISCUSSION THREAD

**Discussion by sabikrahat** [pocketbase/pocketbase#1874] [Q&A] [answered]
Hello!
Whenever I tried to store a file in user avatar field. It shows a error like this....

Error:
`ClientException: {url: http://127.0.0.1:8090/api/collections/users/records, isAbort: false, statusCode: 400, response: {code: 400, message: Failed to load the submitted data due to invalid formatting., data: {avatar: {code: validation_unknown_filenames, message: The field contains unknown filenames.}}}, originalError: null}`

The function was:
```
Future<bool> pocketbaseSignup(PocketBaseUser user, File? img) async {
  final pb = PocketBase(baseUrl);

  debugPrint('Registering User: ${user.toJson()}');
  debugPrint('Registering User Image: ${img?.path}');

  try {
    final res = await pb.collection(users).create(
          body: user.toJson(),
          files: img == null
              ? []
              : [
                  await http.MultipartFile.fromPath(
                    'avatar',
                    img.path,
                    filename: '${user.username}.png',
                  ),
                ],
        );
    log.i('User Creation: $res');
  } catch (e) {
    log.e('User Creation: $e');
    showAwesomeSnackbar('Failed!', '$e', MessageType.failure);
    return false;
  }
  return true;
}
```

**Comment by ganigeorgiev** [comment, marked as answer]
What is the content of  `user.toJson()`?

The `validation_unknown_filenames` error is usually thrown if the user has specified a new file field value, which is not allowed since the file field value(s) (aka. the filenames) are generated based on the original filename(s) and set automatically on upload.

**Reply by sabikrahat** [reply]
The `user.json() `
```
flutter: Registering User: {name: ABC, email: abc@gmail.com, avatar: null, username: abc, password: @Abc123, emailVisibility: true, passwordConfirm: @Abc123}
flutter: Registering User Image: C:\Users\sabik\OneDrive\Desktop\Accounting\logos\test-accounting.jpg
```

I didn't create any new file field value.

**Reply by ganigeorgiev** [reply]
This is because of the `avatar: null`  body field. What actually is happening is that you are sending 2 values to the `avatar` field - `"null"` and the `http.MultipartFile`. You can try removing the `avatar` key from the body, aka:

```dart
...
body: user.toJson()..remove("avatar"),
...
```

**Reply by sabikrahat** [reply]
Thank you so much.... It works...
